libs/agno/agno/os/interfaces/a2a/utils.py

```diff
@@ -0,0 +1,445 @@
+    """Map A2A SendMessageRequest to Agno RunInput.
+
+    1. Validate the request
+    2. Process message parts
+    3. Build and return RunInput
+
+    Args:
+        request_body: A2A-valid JSON-RPC request body dict:
+
+        ```json
+        {
+            "jsonrpc": "2.0",
+            "method": "message/send",
+            "id": "id",
+            "params": {
+                "message": {
+                    "messageId": "id",
+                    "role": "user",
+                    "contextId": "id",
+                    "parts": [{"kind": "text", "text": "Hello"}]
+                }
+            }
+        }
+        ```
+
+    Returns:
+        RunInput: The Agno RunInput
+        stream: Wheter we are in stream mode
+    """
+
+    # 1. Validate the request
+    if stream:
+        try:
+            a2a_request = SendStreamingMessageRequest.model_validate(request_body)
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=f"Invalid A2A request: {str(e)}")
+    else:
+        try:
+            a2a_request = SendMessageRequest.model_validate(request_body)
+        except Exception as e:
+            raise HTTPException(status_code=400, detail=f"Invalid A2A request: {str(e)}")
+
+    a2a_message = a2a_request.params.message
+    if a2a_message.role != "user":
+        raise HTTPException(status_code=400, detail="Only user messages are accepted")
+
+    # 2. Process message parts
+    text_parts = []
+    images = []
+    videos = []
+    audios = []
+    files = []
+
+    for part in a2a_message.parts:
+        # Handle message text content
+        if isinstance(part.root, TextPart):
+            text_parts.append(part.root.text)
+
+        # Handle message files
+        elif isinstance(part.root, FilePart):
+            file_data = part.root.file
+            if isinstance(file_data, FileWithUri):
+                if not file_data.mime_type:
+                    continue
+                elif file_data.mime_type.startswith("image/"):
+                    images.append(Image(url=file_data.uri))
+                elif file_data.mime_type.startswith("video/"):
+                    videos.append(Video(url=file_data.uri))
+                elif file_data.mime_type.startswith("audio/"):
+                    audios.append(Audio(url=file_data.uri))
+                else:
+                    files.append(File(url=file_data.uri, mime_type=file_data.mime_type))
+            elif isinstance(file_data, FileWithBytes):
+                if not file_data.mime_type:
+                    continue
+                files.append(File(content=file_data.bytes, mime_type=file_data.mime_type))
+
+        # Handle message structured data parts
+        elif isinstance(part.root, DataPart):
+            import json
+
+            text_parts.append(json.dumps(part.root.data))
+
+    # 3. Build and return RunInput
+    complete_input_content = "\n".join(text_parts) if text_parts else ""
+    return RunInput(
```

Comment:
Kind of weird to use `RunInput` here since it is really used for other things... but also why not!